contentful/export-processor/content-types/section.js

```diff
@@ -152,6 +152,119 @@ export class Section {
     return minimumPathsForRecommendations;
   }
 
+  /**
+   * Calcutes paths to select each answer to each question
+   */
+
+    getPathsForAllAnswers() {
+        const allAnswerPaths = []
+        const allAnswers = []
+        const answersUsed = []
+
+        // Log ids of all answers in section
+        this.questions.forEach((question) => {
+            question.answers.forEach((answer) => {
+                allAnswers.push(answer.id)
+            })
+        })
+
+        // Log ids of answers used in minimum paths for recommendations
+        const allRecommendationPaths = this.minimumPathsForRecommendations;
+        for (const maturity in allRecommendationPaths) {
+            allRecommendationPaths[maturity].forEach((question) => {
+                if (!answersUsed.includes(question.answer.id)) {
+                    answersUsed.push(question.answer.id)
+                }
+            })
+        }
+
+        // Create paths until all answers are used
+        while (answersUsed.length !== allAnswers.length) {
+            const neededNextQuestions = [];
+            const newPath = [];
+            const lastQuestion = this.questions[this.questions.length - 1];
+            let currentAnswer;
+
+            // Add last question and answer
+            const unusedLastAnswers = lastQuestion.answers.filter(answer => !answersUsed.includes(answer.id))
+
+            if (unusedLastAnswers.length) {
+                newPath.unshift({ question: lastQuestion, answer: unusedLastAnswers[0] })
+                answersUsed.push(unusedLastAnswers[0].id)
+                neededNextQuestions.unshift(lastQuestion.id);
+            } else {
+                newPath.unshift({ question: lastQuestion, answer: lastQuestion.answers[0] })
+            }
+
+            // Loop through remaining questions
+            for (let i = this.questions.length - 2; i >= 0; i--) {
+                const currentQuestion = this.questions[i]
+                if (neededNextQuestions.length) {
+                    const validAnswers = currentQuestion.answers.filter(answer => answer.nextQuestion && answer.nextQuestion.sys.id == neededNextQuestions[0])
+                    const validUnusedAnswers = validAnswers.filter(answer => !answersUsed.includes(answer.id))
+
+                    currentAnswer = validUnusedAnswers.length ? validUnusedAnswers[0] : validAnswers[0]
+
+                    if (validUnusedAnswers.length) {
+                        answersUsed.push(currentAnswer.id)
+                    }
+
+                    neededNextQuestions.unshift(currentQuestion.id)
+
+                } else {
+                    const unusedAnswers = currentQuestion.answers.filter(answer => !answersUsed.includes(answer.id))
+
+                    currentAnswer = unusedAnswers.length ? unusedAnswers[0] : currentQuestion.answers[0]
```

Comment:
I think you could do your checks here as to whether `currentAnswer` has a next question, and if it doesn't splice the path, to save iterating over them all again in another loop at the end

Reply:
Would like to hear your thoughts on how this could work, I think I went the way I did because of the Q4 -> Q6 scenario (which led me down the array route, idea being to find the next truly-needed answer. But could probably do this for !nextQuestion answers at this point.